keystore.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+package pb;
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+option (gogoproto.gostring_all) = true;
+option (gogoproto.equal_all) = true;
+option (gogoproto.verbose_equal_all) = true;
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all) =  true;
+option (gogoproto.populate_all) = true;
+option (gogoproto.testgen_all) = true;
+// uncomment to enable bench generation
+// option (gogoproto.benchgen_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+
+// KeystoreAPI provides a keystore management API
+service KeystoreAPI {
+    // KeystoreHas is used to check if we have the key in our store
+    rpc KeystoreHas(KeystoreRequest) returns (KeystoreResponse) { };
+    // KeystoreGet is used to return a key from our store
+    rpc KeystoreGet(KeystoreRequest) returns (KeystoreResponse) { };
+    // KeystorePut is used to store a key in our store
+    rpc KeystorePut(KeystoreRequest) returns (KeystoreResponse) { };
+    // KeystoreDelete is used to remove a key from our store
+    rpc KeystoreDelete(KeystoreRequest) returns (KeystoreResponse) { };
+    // KeystoreList is used to returns all keyIDs of keys in our store
+    rpc KeystoreList(KeystoreRequest) returns (KeystoreResponse) { };
```

Comment:
What do you put in KeystoreRequest for KeystoreList?

I think the API can be made more clear if we use more request and response types.

Reply:
at the moment nothing, however in the future the request message can include options for limiting list results. Since we're using protocol buffers we can add new fields in the future without breaking api compatability.